### PR TITLE
qgeoview: add qgeoview/1.0.4

### PR DESCRIPTION
**Pull request by Jihadist**
Specify library name and version:  **qgeoview/1.0.4**

QGeoView recipe forces shared library https://github.com/AmonRaNet/QGeoView/blob/master/lib/CMakeLists.txt#L19 so I removed shared option

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/qgeoview/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            del self.options.fPIC
+
+    def configure(self):
+        self.options.rm_safe("fPIC")
+        # FIXME: we shouldn't have to force shared in qt, but config file
+        #        generated by conan in qt static is likely broken, or maybe env vars.
+        self.options["qt"].shared = True
+
+    def requirements(self):
+        self.requires("qt/[>=5.15.2 <7]")
```
Comment (RubenRBS) [inline recipes/qgeoview/all/conanfile.py:38]: Hi! Thanks for the idea, but QT is not yet one of the allowed libraries to be required with version ranges - We're working towards that future, but understanding the implications for each library is a slow process :) Please pin this for now with the version that makes most sense, usually the latest available one in CCI

File: recipes/qgeoview/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        self.options.rm_safe("fPIC")
+        # FIXME: we shouldn't have to force shared in qt, but config file
+        #        generated by conan in qt static is likely broken, or maybe env vars.
```
Comment (RubenRBS) [inline recipes/qgeoview/all/conanfile.py:34]: If this is a bug, could you report an issue to the client/this repo, depending on where you feel it fits better? Thanks!

File: recipes/qgeoview/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+
+class QGeoViewConan(ConanFile):
+    name = "qgeoview"
+    description = "QGeoView is a Qt / C ++ widget for visualizing geographic data. "
+    license = "LGPL-3.0"
+    topics = ("geo", "geospatial", "qt")
+    homepage = "https://github.com/AmonRaNet/QGeoView"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
```
Comment (RubenRBS) [inline recipes/qgeoview/all/conanfile.py:19]: missing `package_type` attribute. [Docs here](https://docs.conan.io/2/reference/conanfile/attributes.html#package-type)

**Review by RubenRBS — CHANGES_REQUESTED** [review]
Thanks a lot! I have a request due to QT's version range usage and a question about a comment in the code, but otherwise the recipe looks good

File: recipes/qgeoview/all/conanfile.py (hunk 1 above)
Comment (Jihadist) [reply recipes/qgeoview/all/conanfile.py:38]: Hi! Thank you for your attention. I tried to fix qt5 or qt6. Conan v1 pipeline works with qt5 but v2 pipeline works only with qt6.

File: recipes/qgeoview/all/conanfile.py (hunk 2 above)
Comment (Jihadist) [reply recipes/qgeoview/all/conanfile.py:34]: I cannot build any qt gui library with qt static and conan v1. It's an old problem that was fixed by v2.

File: recipes/qgeoview/all/conanfile.py (hunk 3 above)
Comment (Jihadist) [reply recipes/qgeoview/all/conanfile.py:20]: done

**Comment by Jihadist** [comment]
Blocked by #22788

**Review by valgur — APPROVED** [review]

**Comment by Jihadist** [comment]
thank you for no reviews ^)

**Pull request closed without merging**